src/basic/list.h

```diff
@@ -1,8 +1,6 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
-#include "macro.h"
```

Comment:
Why this is dropped?

Reply:
We get compilation failures if we leave it in:

```
➜  systemd git:(log-context) ninja -C build
ninja: Entering directory `build'
[165/2017] Compiling C object src/libsystemd/libsystemd_static.a.p/sd-netlink_netlink-types.c.o
FAILED: src/libsystemd/libsystemd_static.a.p/sd-netlink_netlink-types.c.o 
cc -Isrc/libsystemd/libsystemd_static.a.p -Isrc/libsystemd -I../src/libsystemd -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wno-format-signedness -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wunused-function -Wwrite-strings -Wno-unused-result -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -fno-omit-frame-pointer -Werror=shadow -include config.h -fPIC -pthread -fvisibility=default -MD -MQ src/libsystemd/libsystemd_static.a.p/sd-netlink_netlink-types.c.o -MF src/libsystemd/libsystemd_static.a.p/sd-netlink_netlink-types.c.o.d -o src/libsystemd/libsystemd_static.a.p/sd-netlink_netlink-types.c.o -c ../src/libsystemd/sd-netlink/netlink-types.c
In file included from ../src/basic/macro.h:452,
                 from ../src/basic/list.h:4,
                 from ../src/libsystemd/sd-netlink/netlink-internal.h:8,
                 from ../src/libsystemd/sd-netlink/netlink-types.c:6:
../src/basic/log.h:405:9: error: expected specifier-qualifier-list before ‘LIST_FIELDS’
  405 |         LIST_FIELDS(struct log_context, ll);
      |         ^~~~~~~~~~~
../src/basic/log.h:408:51: error: unknown type name ‘_log_context’
  408 | extern thread_local LIST_HEAD(struct log_context, _log_context);
      |                                                   ^~~~~~~~~~~~
../src/basic/log.h: In function ‘_log_context_pop’:
../src/basic/log.h:416:9: error: implicit declaration of function ‘LIST_REMOVE’ [-Werror=implicit-function-declaration]
  416 |         LIST_REMOVE(ll, _log_context, *c);
      |         ^~~~~~~~~~~
../src/basic/log.h:416:9: warning: nested extern declaration of ‘LIST_REMOVE’ [-Wnested-externs]
../src/basic/log.h:416:21: error: ‘ll’ undeclared (first use in this function)
  416 |         LIST_REMOVE(ll, _log_context, *c);
      |                     ^~
../src/basic/log.h:416:21: note: each undeclared identifier is reported only once for each function it appears in
../src/basic/log.h:416:25: error: ‘_log_context’ undeclared (first use in this function); did you mean ‘log_context’?
  416 |         LIST_REMOVE(ll, _log_context, *c);
      |                         ^~~~~~~~~~~~
      |                         log_context
cc1: some warnings being treated as errors
[174/2017] Compiling C object src/libsystemd/libsystemd_static.a.p/sd-resolve_sd-resolve.c.o
ninja: build stopped: subcommand failed.
```

Reply:
How about to include macro.h at the end of the file, like macro.h includes logs.h? Otherwise, we cannot safely use the macros defined in list.h without separately including macro.h.